terra-workspace-manager/sa.tf

```
@@ -1,16 +1,30 @@
-resource "google_service_account" "app" {
+resource "google_service_account" "sqlproxy" {
   count = var.enable ? 1 : 0
 
   provider     = google.target
   project      = var.google_project
-  account_id   = "${local.service}-${local.owner}"
-  display_name = "${local.service}-${local.owner}"
+  account_id   = "${local.service}-${local.owner}-sqlproxy"
+  display_name = "${local.service}-${local.owner}-sqlproxy"
 }
 resource "google_project_iam_member" "app_roles" {
-  count = var.enable ? length(local.sa_roles) : 0
-
   provider = google.target
   project  = var.google_project
-  role     = local.sa_roles[count.index]
-  member   = "serviceAccount:${google_service_account.app[0].email}"
+  role     = "roles/cloudsql.client"
+  member   = "serviceAccount:${google_service_account.sqlproxy[0].email}"
+}
+
+resource "google_service_account" "cloud_trace" {
+  count = var.enable ? 1 : 0
+
+  provider     = google.target
+  project      = var.google_project
+  account_id   = "${local.service}-${local.owner}-cloud-trace"
+  display_name = "${local.service}-${local.owner}-cloud-trace"
 }
+
+resource "google_project_iam_member" "cloud_trace_roles" {
```

Add toggle on enabled
```suggestion
resource "google_project_iam_member" "cloud_trace_roles" {
  count = var.enable ? 1 : 0

```